atomic_reactor/plugin.py

```diff
@@ -354,6 +381,57 @@ def __init__(self, dt, workflow, plugins_conf, *args, **kwargs):
         self.plugins_results = workflow.prebuild_results
         super(PreBuildPluginsRunner, self).__init__(dt, workflow, 'PreBuildPlugin', plugins_conf, *args, **kwargs)
 
+
+class BuildStepPlugin(BuildPlugin):
+    pass
+
+
+class BuildStepPluginsRunner(BuildPluginsRunner):
+
+    def __init__(self, dt, workflow, plugin_conf, *args, **kwargs):
+        self._build_result = None
+
+        logger.info("initializing runner of build-step plugin")
+        self.plugins_results = workflow.buildstep_result
+
+        if not plugin_conf:
+            plugin_conf = [{'name': 'docker_api'}]
+        else:
+            for i in range(len(plugin_conf)):
+                plugin_conf[i]['required'] = False
+
+        super(BuildStepPluginsRunner, self).__init__(
+            dt, workflow, 'BuildStepPlugin', plugin_conf, *args, **kwargs)
+
+    def run(self, *args, **kwargs):
+        builder = self.workflow.builder
+
+        logger.info('building image %r inside current environment',
+                    builder.image)
+        builder._ensure_not_built()
+        logger.debug('using dockerfile:\n%s',
+                     DockerfileParser(builder.df_path).content)
+
+        plugins_results = super(BuildStepPluginsRunner, self).run(*args, **kwargs)
```

Comment:
This invocation should supply buildstep_phase=True to the inherited class's run(), instead of having callers of this run() have to supply it.